cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -372,6 +379,19 @@ public E addResponseInterceptors(List<HttpConnectionResponseInterceptor> interce
         return (E) this;
     }
 
+    public E username(String username) {
+        this.username = username;
```

Comment:
I'm not sure, it doesn't exactly breaks anything if you pass in `null`. It could be useful in some circumstances to allow `null`, but I don't really have a strong opinion either way on this one.